app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/MainActivityTest.kt

```diff
@@ -416,4 +422,64 @@ class MainActivityTest {
             }
         }
     }
+
+    @Test
+    fun clickOnCapturePhotoDontThrowsError() {
+
+        activityRule.scenario.onActivity {
+            val connectedUserViewModel by it.viewModels<ConnectedUserViewModel>()
+            connectedUserViewModel.setCurrentUser(user1)
+        }
+        onView(withId(R.id.camera_capture_button)).perform(click())
+        //BaseRobot().assertOnView(withId(R.id.sendPostFragment), matches(isDisplayed()))
+
+    }
```

Comment:
I think this test could be moved to `CameraWithUIFragmentTest`. It doesn't really make sens to test it from the `MainActivity`

And to test that it doesn't throw an error I would rather write it like that:

```suggestion
@Test
    fun clickOnCapturePhotoDontThrowsError() {
        try {
                onView(withId(R.id.camera_capture_button)).check(matches(isDisplayed())).perform(click())
        } catch (e: Exception) {
                fail("An exception has been thrown ${e.message}")
        }
    }
```

Reply:
Ok, I see. Yes it behaves a bit strangely. I don't really understand it either. It looks all good for me to let it here. Thanks a lot for your hard work at trying to test this difficult part, well done !! :+1: